packages/block-library/src/tag-cloud/edit.js

```diff
@@ -54,6 +78,33 @@ function TagCloudEdit( { attributes, setAttributes, taxonomies } ) {
 		return [ selectOption, ...taxonomyOptions ];
 	};
 
+	const onFontSizeChange = ( fontSizeLabel, newValue ) => {
+		// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+		const [ quantity, newUnit ] = parseQuantityAndUnitFromRawValue(
+			newValue
+		);
+		if ( ! Number.isFinite( quantity ) || quantity < 0 || quantity > 100 ) {
```

Comment:
With #39186 merged, would you be able to rebase on on top of latest `trunk` and see if the manual `quantity < 0 || quantity > 100` check can now swapped with `event.target?.validity.valid` (see [docs](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/number-control/README.md?plain=1#L100-L112))?

Reply:
`UnitControl` uses `NumberControl` under the hood — what I'm trying to say is that we should now able to use `min`/`max` instead of this manual check:

<details>
<summary>Click to expand</summary>


```diff
diff --git a/packages/block-library/src/tag-cloud/edit.js b/packages/block-library/src/tag-cloud/edit.js
index ae52d42f13..23515f1764 100644
--- a/packages/block-library/src/tag-cloud/edit.js
+++ b/packages/block-library/src/tag-cloud/edit.js
@@ -41,6 +41,9 @@ const MIN_TAGS = 1;
  */
 const MAX_TAGS = 100;
 
+const MIN_FONT_SIZE = 1;
+const MAX_FONT_SIZE = 100;
+
 function TagCloudEdit( { attributes, setAttributes, taxonomies } ) {
 	const {
 		taxonomy,
@@ -83,7 +86,7 @@ function TagCloudEdit( { attributes, setAttributes, taxonomies } ) {
 		const [ quantity, newUnit ] = parseQuantityAndUnitFromRawValue(
 			newValue
 		);
-		if ( ! Number.isFinite( quantity ) || quantity < 0 || quantity > 100 ) {
+		if ( ! Number.isFinite( quantity ) ) {
 			return;
 		}
 		const updateObj = { [ fontSizeLabel ]: newValue };
@@ -142,6 +145,8 @@ function TagCloudEdit( { attributes, setAttributes, taxonomies } ) {
 								onFontSizeChange( 'smallestFontSize', value );
 							} }
 							units={ units }
+							min={ MIN_FONT_SIZE }
+							max={ MAX_FONT_SIZE }
 						/>
 					</FlexItem>
 					<FlexItem isBlock>
@@ -152,6 +157,8 @@ function TagCloudEdit( { attributes, setAttributes, taxonomies } ) {
 								onFontSizeChange( 'largestFontSize', value );
 							} }
 							units={ units }
+							min={ MIN_FONT_SIZE }
+							max={ MAX_FONT_SIZE }
 						/>
 					</FlexItem>
 				</Flex>
```

</details>

Reply:
sorry I misunderstood your point. Let me try this ☝️